From 486ea724ef94ada14b876f07e28ebd73b9ae63b9 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <i.gnatenko.brain@gmail.com>
Date: Wed, 31 Aug 2016 17:30:56 +0200
Subject: [PATCH] build: don't set default build type and define _GNU_SOURCE

We will let CMake handle default build type. Also remove all custom
CFLAGS and use system ones.

References: https://bugzilla.redhat.com/show_bug.cgi?id=1332067
Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>
---
 CMakeLists.txt | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 06432d3..4d9ef5b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,11 +5,13 @@ CMAKE_POLICY (SET CMP0005 OLD)
 INCLUDE (${CMAKE_SOURCE_DIR}/VERSION.cmake)
 SET (VERSION "${HAWKEY_MAJOR}.${HAWKEY_MINOR}.${HAWKEY_PATCH}")
 
-set (CMAKE_C_FLAGS		"${CMAKE_C_FLAGS} -std=c99 -Werror=implicit-function-declaration -Wall -Wl,--as-needed")
-set (CMAKE_C_FLAGS_DEBUG	"${CMAKE_C_FLAGS} -ggdb -O0")
-IF(NOT CMAKE_BUILD_TYPE)
-  set (CMAKE_BUILD_TYPE DEBUG)
-ENDIF(NOT CMAKE_BUILD_TYPE)
+include (CheckSymbolExists)
+list (APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
+check_symbol_exists (FNM_CASEFOLD "fnmatch.h" HAS_FNM_CASEFOLD)
+if (NOT HAS_FNM_CASEFOLD)
+  message (SEND_ERROR "FNM_CASEFOLD is not available")
+endif ()
+add_definitions (-D_GNU_SOURCE)
 
 INCLUDE_DIRECTORIES (${CMAKE_SOURCE_DIR})
 
-- 
2.9.3

